gapic/templates/%namespace/%name_%version/%sub/services/%service/client.py.j2

```diff
@@ -79,15 +82,13 @@ class {{ service.client_name }}(metaclass={{ service.client_name }}Meta):
 
 
     def __init__(self, *,
-            host: str{% if service.host %} = '{{ service.host }}'{% endif %},
```

Comment:
This change is a bit scary. We may have situations where the `host` may not be specified (for instance, APIs with host-based regionalization). I made the original change to make overriding `host` not feel like a gymnastics exercise.

Also, the previous code had `host` as a required argument if we did not have a sensible default, but this change loses that.

Reply:
Again, preserving the legacy interface is preventing us from adopting better semantics.
@busunkim96 indicated that more configuration may be added to `client_options` in the future, and having the data structure augmented is arguably better than bloating the constructor signature.
I didn't know about APIs being able to not set a host. The new change should still catch that, no? If host is a required parameter, static analysis can detect its absence, but the transport will still raise if it isn't given a host. It's a minor UI wart, but the user is till prevented from constructing a client sans host.